com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Filter/DitherNode.cs

```diff
@@ -0,0 +1,55 @@
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+
+    internal class DitherNode : IStandardNode
+    {
+        public static string Name = "Dither";
+        public static int Version = 1;
+
+        public static FunctionDescriptor FunctionDescriptor => new(
+            Version,
+            Name,
+@"
+{
+    uv = ScreenPosition.xy * _ScreenParams.xy;
+    uint index = (uint(uv.x) % 4) * 4 + uint(uv.y) % 4;
+    Out = In - DITHER_THRESHOLDS[index];
+}",
+            new ParameterDescriptor("In", TYPE.Vector, Usage.In),
+            new ParameterDescriptor("ScreenPosition", TYPE.Vec2, Usage.In),//TODO: ScreenPosition default support
+            new ParameterDescriptor("Out", TYPE.Vector, Usage.Out),
+            new ParameterDescriptor("uv", TYPE.Vec2, Usage.Local),
+            new ParameterDescriptor("DITHER_THRESHOLDS", TYPE.Mat4, Usage.Local, new float[]
+        {
+        1.0f / 17.0f,  9.0f / 17.0f,  3.0f / 17.0f, 11.0f / 17.0f,
+        13.0f / 17.0f,  5.0f / 17.0f, 15.0f / 17.0f,  7.0f / 17.0f,
+        4.0f / 17.0f, 12.0f / 17.0f,  2.0f / 17.0f, 10.0f / 17.0f,
+        16.0f / 17.0f,  8.0f / 17.0f, 14.0f / 17.0f,  6.0f / 17.0f
+        })
+        );
+
+        public static NodeUIDescriptor NodeUIDescriptor => new(
```

Comment:
Seems like there's some delta between the preview output for dither node in SG 1.0 and SG 2.0, the newer preview output looks kinda red tinted 
![image](https://user-images.githubusercontent.com/13033542/167043843-e67b2b18-5682-4ebb-a203-d8b07bc1aff1.png)

Reply:
This could be a preview related issue, may not have anything to do with the node definition, just calling it out

Reply:
this is a bit weird, `Out = In - DITHER_THRESHOLDS[index];` results on a tinted color but if I do `    
Out.x = In.x - DITHER_THRESHOLDS[index];
    Out.y = In.y - DITHER_THRESHOLDS[index];
    Out.z = In.z - DITHER_THRESHOLDS[index];` then the result will be grayscale. I am not sure what's causing the tinted result 🤔 But I guess we will want to keep `Out = In - DITHER_THRESHOLDS[index];` since  the input could be any size of the vectors.

Reply:
Is there a possibility that it might be a preview issue?

Reply:
IMO we should wait to address this until the real dynamic vector system is in place.

Reply:
I'll take a note to look at this next sprint @tracychen1234

Reply:
And yeah I guess it makes sense to wait a bit Ben, until the node is fully featured